src/sentry/integrations/utils/metrics.py

```diff
@@ -197,3 +203,48 @@ def __exit__(
                 if self.assume_success
                 else EventLifecycleOutcome.HALTED
             )
+
+
+class IntegrationPipelineViewType(Enum):
+    """A specific step in an integration's pipeline that is not a static page."""
+
+    # IdentityProviderPipeline
+    IDENTITY_PROVIDER = "IDENTITY_PROVIDER"
+
+    # GitHub
+    OAUTH_LOGIN = "OAUTH_LOGIN"
+    GITHUB_INSTALLATION = "GITHUB_INSTALLATION"
+
+    # Bitbucket
+    VERIFY_INSTALLATION = "VERIFY_INSTALLATION"
+
+    # Bitbucket Server
+    # OAUTH_LOGIN = "OAUTH_LOGIN"
+    OAUTH_CALLBACK = "OAUTH_CALLBACK"
+
+    # Azure DevOps
+    ACCOUNT_CONFIG = "ACCOUNT_CONFIG"
+
+    def __str__(self) -> str:
+        return self.value.lower()
+
+
+@dataclass
+class IntegrationPipelineViewEvent(EventLifecycleMetric):
+    """An instance to be recorded of a user going through an integration pipeline view (step)."""
+
+    interaction_type: IntegrationPipelineViewType
+    domain: str
```

Comment:
Could we start an `Enum` class for domain values as well? It's fine if `"source_code"` is the only value to start with.

Reply:
coming up in a follow up!